Public/Src/Cache/ContentStore/Library/Stores/FileSystemContentStoreInternalChecker.cs

```diff
@@ -235,15 +234,15 @@ int findNextIndexToProcess(ContentHash lastProcessedHash)
 
             void traceProgressIfNeeded(ContentHash currentHash, int currentHashIndex)
             {
-                processedBytes += contentHashes[currentHashIndex].fileInfo.Length;
+                processedBytes += contentHashes[currentHashIndex].Payload.Length;
 
                 var swTime = stopwatch.Elapsed;
                 if (swTime - progressTracker > _settings.SelfCheckProgressReportingInterval)
                 {
                     // It is possible to have multiple replicas with the same hash.
                     // We need to save the state only when *all* the replicas are processed.
                     // So we check if the next item has a different hash.
-                    if (currentHash != contentHashes[currentHashIndex + 1].hash)
+                    if (currentHash != contentHashes[currentHashIndex + 1].Hash)

```

Comment:
What happens when currentHashIndex points to last element? (I know this + 1 was in code already, but I would like to be convinced of safety while we are looking at this code.)

Reply:
I'm not sure why it was being done before, but it certainly seems like an error that did not trigger (probably something to do with the reporting intervals).

I have added a check to ensure we don't enter it at the last index, as the state will be saved immediately afterwards.